transport/transport.go

```diff
@@ -81,6 +81,18 @@ type Transport interface {
 	// See the Network interface for an explanation of how this is used.
 	// TODO: Make this a part of the go-multiaddr protocol instead?
 	Proxy() bool
+	
+	// If your transport implements `io.Closer` Close will be called 
+	// when libp2p doesn't need it anymore.
+	// (libp2p closing, transport unhooked from a node, ...)
+	//
+	// Once a Close is called you can safely assume no more Dial or Listen are
+	// gonna be issued on this instance.
+	//
+	// If Close return an error the transport is still considered as dead
+	// (ex: transport already closed or transport never started).
+	//
+	// No more than one Close will ever be issued on an instance.
```

Comment:
This is implied in the contract. (or we can extend the statement above to be "If your transport implements `io.Closer` Close will be called exactly once when the libp2p node shuts down".